staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go

```diff
@@ -229,3 +235,24 @@ func withAuthorization(validate rest.ValidateObjectFunc, a authorizer.Authorizer
 		return errors.NewForbidden(gr, name, err)
 	}
 }
+
+// updateToCreateOptions creates a CreateOptions with the same field values as the provided UpdateOptions.
+func updateToCreateOptions(po *metav1.UpdateOptions) (*metav1.CreateOptions, error) {
```

Comment:
directly copying fields in these methods would be way more performant, e.g.:

```go
if updateOptions == nil {
  return &metav1.CreateOptions{}, nil
}
return &metav1.CreateOptions{
  DryRun: updateOptions.DryRun,
  FieldManager: updateOptions.FieldManager,
}, nil
```

unit tests that fuzz then do an object -> json -> object transfer to make sure we caught all equivalent fields would make sense

Reply:
+1

Reply:
Yes.. I mindlessly copy-pasted this approach over from the pre-existing `handlers.updateToCreateOptions()` function, but I shouldn't have. I'll fix them and add a test to ensure we don't end up dropping fields in the future.